src/Dictionaries/HashedDictionary.cpp

```diff
@@ -1,271 +1,440 @@
-        extractor);
-}
+    // if constexpr (dictionary_key_type == DictionaryKeyType::simple)
+    // {
+    //     ColumnPtr result;
 
+    //     PaddedPODArray<KeyType> backup_storage;
+    //     const auto & ids = getColumnVectorData(this, key_columns.front(), backup_storage);
 
-/// Allow to use single value in same way as array.
-static inline HashedDictionary::Key getAt(const PaddedPODArray<HashedDictionary::Key> & arr, const size_t idx)
-{
-    return arr[idx];
-}
-static inline HashedDictionary::Key getAt(const HashedDictionary::Key & value, const size_t)
-{
-    return value;
-}
+    //     auto size = ids.size();
 
-template <typename AttrType, typename ChildType, typename AncestorType>
-void HashedDictionary::isInAttrImpl(const AttrType & attr, const ChildType & child_ids, const AncestorType & ancestor_ids, PaddedPODArray<UInt8> & out) const
-{
-    const auto null_value = std::get<UInt64>(hierarchical_attribute->null_values);
-    const auto rows = out.size();
+    //     const auto & dictionary_attribute = dict_struct.getAttribute(attribute_name, result_type);
+    //     DefaultValueProvider default_value_provider(dictionary_attribute.null_value, default_values_column);
 
-    for (const auto row : ext::range(0, rows))
-    {
-        auto id = getAt(child_ids, row);
-        const auto ancestor_id = getAt(ancestor_ids, row);
+    //     size_t index = dict_struct.attribute_name_to_index.find(attribute_name)->second;
+    //     const auto & attribute = attributes[index];
 
-        for (size_t i = 0; id != null_value && id != ancestor_id && i < DBMS_HIERARCHICAL_DICTIONARY_MAX_DEPTH; ++i)
-        {
-            auto it = attr.find(id);
-            if (it != std::end(attr))
-                id = second(*it);
-            else
-                break;
-        }
+    //     auto type_call = [&](const auto & dictionary_attribute_type)
+    //     {
+    //         using Type = std::decay_t<decltype(dictionary_attribute_type)>;
+    //         using AttributeType = typename Type::AttributeType;
+    //         using ValueType = DictionaryValueType<AttributeType>;
+    //         using ColumnProvider = DictionaryAttributeColumnProvider<AttributeType>;
 
-        out[row] = id != null_value && id == ancestor_id;
-    }
+    //         auto & container = std::get<CollectionType<ValueType>>(attribute.container);
 
-    query_count.fetch_add(rows, std::memory_order_relaxed);
-}
-template <typename ChildType, typename AncestorType>
-void HashedDictionary::isInImpl(const ChildType & child_ids, const AncestorType & ancestor_ids, PaddedPODArray<UInt8> & out) const
-{
-    if (!sparse)
-        return isInAttrImpl(*std::get<CollectionPtrType<Key>>(hierarchical_attribute->maps), child_ids, ancestor_ids, out);
-    return isInAttrImpl(*std::get<SparseCollectionPtrType<Key>>(hierarchical_attribute->sparse_maps), child_ids, ancestor_ids, out);
-}
+    //         auto column = ColumnProvider::getColumn(dictionary_attribute, size);
```

Comment:
?

Reply:
I commented previous imlementation to check if there is no performance degradation.

Reply:
I will create performance tests to check if there was no regression with these changes.